File: coursework/final/clean_names-unedited.py
# clean_names.py
# all code can be found at: https://github.com/felttipfaerie/Launch-Into-Computing/tree/master/coursework/final

def process_names(names):
    unique_names = list(set(names))  # removes duplicates but doesn't preserve order
    unique_names.sort()  # sorts alphabetically
    return unique_names

# Test list with duplicates and inconsistent capitalisation
test_names = ["Alice", "bob", "alice", "Charlie", "BOB", "dave", "Eve", "charlie"]

print("Original list:", test_names)
print("Processed list:", process_names(test_names))

File: coursework/final/cleanNames-unedited.js
// cleanNames.js
// all code can be found at: https://github.com/felttipfaerie/Launch-Into-Computing/tree/master/coursework/final

function processNames(names) {
    let uniqueNames = [...new Set(names)]; // removes duplicates
    uniqueNames.sort(); // sorts alphabetically
    return uniqueNames;
}

// Test list with duplicates and inconsistent capitalisation
const testNames = ["Alice", "bob", "alice", "Charlie", "BOB", "dave", "Eve", "charlie"];

console.log("Original list:", testNames);
console.log("Processed list:", processNames(testNames));
